### ISSUE TITLE
nan loss at the beginning of training process

### ISSUE THREAD

**Issue by ShixuanGu** [MIC-DKFZ/nnUNet#1288]
Hi, thanks for the great work!
I ran into the issue of nan train/validation loss at the 1st epoch when train models (3D_fullres, 2D) on RibSeg dataset. (Task156_RibSeg)

I haven't figured it out, would you please give a few potential causes for this issue?

To roll out the possibilities, I list a few potential problems here:

- in dataset.json file, the tensorImageSize is "4D", although the input image is 3D nifiti files. (I also change it to 3D and run the training code, the loss is still nan)
- when running **nnUNet_plan_and_preprocess** with **--verify_dataset_integrity**, there are a few lines of "no separate z, order 1/2/3" going on, but in the final line it shows "normalization done".
- I guess another problem is the training data, I didn't use the source CT as input. Instead, I binarized the CT and filtered some non-bone regions, and use the binary CT volume as input (the file size is smaller, int8). Currently, I'm trying to run the test on the source image.

**Comment by FabianIsensee** [comment]
Hey, 
so it's hard to say what could cause this. I know we have successfully trained the ribseg dataset before, specifically @Karol-G. Maybe he can chime in?

>in dataset.json file, the tensorImageSize is "4D", although the input image is 3D nifiti files. (I also change it to 3D and run the training code, the loss is still nan)

this is ignored and doesnt matter

> when running nnUNet_plan_and_preprocess with --verify_dataset_integrity, there are a few lines of "no separate z, order 1/2/3" going on, but in the final line it shows "normalization done".

this is just FYI and is not an error

> I guess another problem is the training data, I didn't use the source CT as input. Instead, I binarized the CT and filtered some non-bone regions, and use the binary CT volume as input (the file size is smaller, int8). Currently, I'm trying to run the test on the source image.

in principle this should also work but why not first try the original images and then start fiddling with things?

Best,
Fabian

**Comment by ShixuanGu** [comment]
Thanks, the original images as input works!
